mypyc/ir/rtypes.py

```diff
@@ -436,6 +443,46 @@ def deserialize(cls, data: JsonDict, ctx: 'DeserMaps') -> 'RTuple':
 )
 
 
+class RStruct(RType):
+    """
+    """
+    def __init__(self,
+                 name: str,
+                 names: List[str],
+                 types: List[RType],
+                 offsets: List[int]) -> None:
```

Comment:
The offsets can be inferred from types of fields. First, we can calculate the size of each simple RType easily (e.g. 32-bit integer -> 4 bytes). Second, we need to align each field based on platform alignment rules. The information in https://en.wikipedia.org/wiki/Data_structure_alignment should do the trick. 

Finally, we need to be able to calculate the alignment and the size of the struct, since structs can be placed inside other structs. The previous link will explain this as well.